WooCommerce/Classes/ViewRelated/Orders/Order Creation/CustomerSection/OrderCustomerSection.swift

```diff
@@ -0,0 +1,141 @@
+                    }
+                }
+            }
+            .padding(.horizontal, insets: geometry.safeAreaInsets)
+            .padding()
+            .background(Color(.listForeground))
+
+            Divider()
+        }
+    }
+
+    var createCustomerView: some View {
+        Button(Localization.addCustomer) {
+            showAddressForm.toggle()
+        }
+        .buttonStyle(PlusButtonStyle())
+        .sheet(isPresented: $showAddressForm) {
+            NavigationView {
+                EditOrderAddressForm(dismiss: {
+                    showAddressForm.toggle()
+                }, viewModel: CreateOrderAddressFormViewModel(siteID: viewModel.siteID,
+                                                              address: viewModel.orderDetails.billingAddress,
+                                                              onAddressUpdate: { updatedAddress in
+                    viewModel.orderDetails.billingAddress = updatedAddress
+                }))
+            }
+        }
+    }
+}
+
+// MARK: Constants
+private extension OrderCustomerSection {
+    enum Layout {
+        static let verticalSpacing: CGFloat = 22.0
+        static let linkButtonTopPadding: CGFloat = 12.0
+        static let linkButtonTrailingPadding: CGFloat = 22.0
+    }
+
+    enum Localization {
+        static let customer = NSLocalizedString("Customer", comment: "Title text of the section that shows Customer details when creating a new order")
+        static let addCustomer = NSLocalizedString("Add customer", comment: "Title text of the button that adds a customer when creating a new order")
+        static let editButton = NSLocalizedString("Edit", comment: "Button to edit a customer on the New Order screen")
+
+        static let billingTitle = NSLocalizedString("Billing Address", comment: "Title for the Billing Address section in order customer data")
+        static let shippingTitle = NSLocalizedString("Shipping Address", comment: "Title for the Edit Shipping Address section in order customer data")
+    }
+}
+
+struct OrderCustomerSection_Previews: PreviewProvider {
+    static var previews: some View {
+        let emptyViewModel = NewOrderViewModel(siteID: 123)
+        let viewModel: NewOrderViewModel = {
+            let vm = NewOrderViewModel(siteID: 123)
+            let sampleAddress = Address(firstName: "Johnny",
+                                        lastName: "Appleseed",
+                                        company: nil,
+                                        address1: "234 70th Street",
+                                        address2: nil,
+                                        city: "Niagara Falls",
+                                        state: "NY",
+                                        postcode: "14304",
+                                        country: "US",
+                                        phone: "333-333-3333",
+                                        email: "scrambled@scrambled.com")
```

Comment:
I generally prefer to not add any model-domain code to our views. Aside to couple them to a layer they have nothing to do with, it opens a window share code between layers without intention. Recently we just [had a problem because](https://github.com/woocommerce/woocommerce-ios/pull/5709) we are using some previews helper on production code, and the source of it was the need to instantiate a VM with a networking entity.

To avoid, you should create your views in a way where they can receive the raw parameters too (good for reusability) what do you think?

Reply:
I see a value here. What about e740d0dd6beb3776a2e959cde8b64541e81c8d13?
In past we also added `#if DEBUG` checks to all SwiftUI previews to prevent unwanted code leaks. Should we actually use it more actively?

Reply:
> In past we also added #if DEBUG checks to all SwiftUI previews to prevent unwanted code leaks. Should we use it more actively?

Actually, that was the problem, any code created inside a preview_provider extension is only compiled for debug modes, but accessible through the whole code base.